### PR TITLE
Implement Tree and Wall toolUpdate Down

### PR DESCRIPTION
**Pull request by duncanspumpkin**

### CODE REVIEW

File: src/OpenLoco/GameCommands/GameCommands.h [hunk 1]
```diff
@@ -433,6 +433,48 @@ namespace OpenLoco::GameCommands
...
+        {
+        }
+
+        Map::Pos2 pos;
+        uint8_t rotation;
+        uint8_t type;
+        uint8_t quadrant;
+        Colour_t colour;
+        bool buildImmediately = false;
+        bool unkFlag = false;
```
Comment (duncanspumpkin) [inline src/OpenLoco/GameCommands/GameCommands.h:456]: `fullClearance`. Prevents trees from being placed under roads/track (used by scatter tools)

File: src/OpenLoco/Math/Trigonometry.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            if (direction & (1 << 12))
+            {
+                value = _quarterSine[(-direction) & 0xFFF];
+            }
+            else
+            {
+                value = _quarterSine[direction & 0xFFF];
+            }
+        }
+        return value * magnitude / 32768;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Math/Trigonometry.cpp:36]: Should we just leave this as 0x8000 instead? Same below. Looks a little cleaner, to me.

Comment (duncanspumpkin) [reply src/OpenLoco/Math/Trigonometry.cpp:36]: Hmm perhaps the other option is `(1<<16)`

File: src/OpenLoco/Windows/TerraForm.cpp [hunk 3]
```diff
@@ -395,27 +408,297 @@ namespace OpenLoco::Ui::Windows::Terraform
...
+                        if (GameCommands::do_23(GameCommands::Flags::apply, *placementArgs) != GameCommands::FAILURE)
+                        {
+                            Audio::playSound(Audio::SoundId::construct, GameCommands::getPosition());
+                        }
+                        break;
+                    case treeCluster::selected:
+                    {
+                        // TODO: Lock behind sandbox/editor
+                        auto previousId = CompanyManager::updatingCompanyId();
+                        CompanyManager::updatingCompanyId(CompanyId::neutral);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Windows/TerraForm.cpp:678]: This is great. Looks like it won't be that much work to make tree clusters cost money, then. Something you'd prefer to tackle in another PR?

Comment (duncanspumpkin) [reply src/OpenLoco/Windows/TerraForm.cpp:678]: Yeah i think so

**Review by AaronVanGeffen — CHANGES_REQUESTED** [review]
Tree (cluster) and wall placement is working nicely, however the primary error texts appear to be missing for both. I just get 'Object in the way' rather than 'Cannot plant this here' (trees) or 'Cannot build this here' (walls).

**Comment by duncanspumpkin** [comment]
Found one other missing error title.

File: src/OpenLoco/Math/Trigonometry.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        if (direction & (1 << 13))
+        {
+            if (direction & (1 << 12))
+            {
+                value = _quarterSine[direction & 0xFFF];
+            }
+            else
+            {
+                value = -_quarterSine[(-direction) & 0xFFF];
+            }
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Math/Trigonometry.cpp:52]: I would prefer to make these blocks a little more compact, e.g.:
```suggestion
            auto sineIndex = (direction & (1 << 12) ? direction : -direction) & 0xFFF;
            value = _quarterSine[sineIndex];
```
Any preference either way?

**Review by AaronVanGeffen — APPROVED** [review]
Tree and wall placement are working properly now. Left a minor comment.

File: src/OpenLoco/Math/Trigonometry.cpp (hunk 4 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Math/Trigonometry.cpp:52]: Unfortunately that is not the same. You have to negate the _quarterSine value on one branch.  Could do that in the sine wave though as that is more symmetrical.

File: src/OpenLoco/Math/Trigonometry.cpp [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#include "Trigonometry.hpp"
+#include "../Interop/Interop.hpp"
+
+namespace OpenLoco::Math::Trigonometry
+{
+    // Excel Function =SIN((A1/16384)*2*PI())*32768
+    // Where A1 is 0 : 4095
+    static Interop::loco_global<int16_t[directionPrecisionHigh / 4], 0x00501B50> _quarterSine;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Math/Trigonometry.cpp:8]: If this is the only place this is used, perhaps we should just integrate the array?

File: src/OpenLoco/Math/Trigonometry.cpp (hunk 4 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/Math/Trigonometry.cpp]: Nice, compact solution! I like it.

File: src/OpenLoco/Math/Trigonometry.cpp (hunk 5 above)
Comment (duncanspumpkin) [reply src/OpenLoco/Math/Trigonometry.cpp:8]: It is the only place so yes would be ripe for integration. Just havent done it yet.

**Pull request merged**